src/main/java/decodex/parser/Parser.java

```diff
@@ -1,5 +1,66 @@
 package decodex.parser;
 
+import decodex.commands.Command;
+import decodex.commands.ExitCommand;
+import decodex.data.DataManager;
+import decodex.modules.ModuleManager;
+import decodex.ui.Ui;
+
+import java.util.Arrays;
+
+/**
+ * Parses and validates the user input.
+ */
 public class Parser {
 
+    /**
+     * Returns the type of command that user has entered.
+     *
+     * @param userInput The input specified by the user.
+     * @return The type of command.
+     */
+    public String getCommandType(String userInput) {
+        String[] tokens = userInput.split(" ");
+        return tokens[0];
+    }
+
+    /**
+     * Returns the Argument portion of the user input.
+     *
+     * @param userInput The input specified by the user.
+     * @return The argument portion of the user input.
+     */
+    public String[] getUserArguments(String userInput) {
+        try {
+            String[] tokens = userInput.split(" ");
+            return Arrays.copyOfRange(tokens,1, tokens.length);
+        } catch (ArrayIndexOutOfBoundsException err) {
+            // Temporary error handling
+            return null;
+        }

```

Comment:
Good catch for the `ArrayIndexOutOfBoundsException` but perhaps let this method throw the exception instead of returning `null`? Then perhaps the exception can be handled in the main loop instead of here.